compiler/ast/types.nim

```diff
@@ -513,27 +499,52 @@ proc lastOrd*(conf: ConfigRef; t: PType): Int128 =
   of tyUInt64:
     result = toInt128(0xFFFFFFFFFFFFFFFF'u64)
   of tyEnum:
-    if t.n.len > 0:
-      assert(t.n[^1].kind == nkSym)
-      result = toInt128(t.n[^1].sym.position)
+    result =
+      if t.n.len > 0:
+        assert(t.n[^1].kind == nkSym)
+        toInt128(t.n[^1].sym.position)
+      else:
+        # currently empty enums aren't support as a type, but they should be
+        Zero
   of tyGenericInst, tyDistinct, tyTypeDesc, tyAlias, tySink,
      tyStatic, tyInferred, tyUserTypeClasses, tyLent:
     result = lastOrd(conf, lastSon(t))
-  of tyProxy: result = Zero
+  of tyProxy: result = Zero # xxx: this seems off; also not in `firstOrd`
   of tyOrdinal:
-    if t.len > 0: result = lastOrd(conf, lastSon(t))
-    else:
-      conf.localReport InternalReport(
-        kind: rintUnreachable,
-        msg: "invalid kind for firstOrd(" & $t.kind & ')')
-  of tyUncheckedArray:
+    conf.internalAssert(t.len > 0, "lastOrd - abstract ordinal type given")
+    result = lastOrd(conf, lastSon(t))
+  of tyUncheckedArray: # xxx: what about `tyCstring`; see `firstOrd`
```

Comment:
Neither `cstring` nor `UncheckedArray` have a compile-time-known lower or upper bound, so I think both shouldn't be handled by `firstOrd`/`lastOrd`.

This is also not something I expect to happen now.

Reply:
Yeah they're pessimisticly `-1`, as in empty; unless we have some hints like it's not `nil`.

Reply:
captured the above as a comment; even if somewhat oblique.